File: CS/06_Annotations/TextAnnotationProperties.cs
using Spire.Pdf;
using Spire.Pdf.Annotations;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace TextAnnotationProperties
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            
            // Create a new PDF document object.
            PdfDocument pdf = new PdfDocument();

            // Load an existing PDF document from a file.
            pdf.LoadFromFile(@"..\..\..\..\..\..\Data\FreeTextAnnotation.pdf");

            // Get the first page of the loaded document.
            PdfPageBase firstPage = pdf.Pages[0];

            // Create a new PDF document object to store the copied annotations.
            PdfDocument newPdf = new PdfDocument();

            // Iterate through each annotation in the first page's annotation list.
            foreach (PdfAnnotation annotation in firstPage.Annotations.List)
            {

                // Check if the annotation is a free text annotation.
                if (annotation is PdfFreeTextAnnotationWidget)
                {

                    // Convert the annotation to a free text annotation object.
                    PdfFreeTextAnnotationWidget textAnnotation = annotation as PdfFreeTextAnnotationWidget;

                    // Retrieve the rectangle bounds of the annotation.
                    var rect = textAnnotation.Bounds;

                    // Retrieve the text content of the annotation.
                    var text = textAnnotation.Text;

                    // Create a new page in the new PDF document with the same size as the first page.
                    PdfPageBase newPage = newPdf.Pages.Add(firstPage.Size);

                    // Create a new free text annotation with the same rectangle bounds.
                    PdfFreeTextAnnotation newAnnotation = new PdfFreeTextAnnotation(rect);

                    // Set the text content of the new annotation.
                    newAnnotation.Text = text;

                    // Copy the callout lines information from the original annotation.
                    newAnnotation.CalloutLines = textAnnotation.CalloutLines;

                    // Copy the line ending style from the original annotation.
                    newAnnotation.LineEndingStyle = textAnnotation.LineEndingStyle;

                    // Set the annotation intent to indicate it's a free text callout.
                    newAnnotation.AnnotationIntent = PdfAnnotationIntent.FreeTextCallout;

                    // Copy the rectangular differences information from the original annotation.
                    newAnnotation.RectangleDifferences = textAnnotation.RectangularDifferenceArray;

                    // Set the color of the new annotation to match the original annotation.
                    newAnnotation.Color = textAnnotation.Color;

                    // Add the new annotation to the annotations widget of the new page.
                    newPage.Annotations.Add(newAnnotation);

                }
            }
            // Save the new PDF document with copied annotation properties to a file.
            String result = "CopyTextAnnotationProperties.pdf";
            newPdf.SaveToFile(result);


            //Launch the file.
            DocumentViewer(result);
        }

        private void DocumentViewer(string filename)
        {
            try
            {
                System.Diagnostics.Process.Start(filename);
            }
            catch { }
        }
    }
}

File: VB/06_Annotations/TextAnnotationProperties.vb
Imports Spire.Pdf
Imports Spire.Pdf.Annotations

Namespace TextAnnotationProperties
    Partial Public Class Form1
        Inherits Form
        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
            ' Create a new PdfDocument object 
            Dim pdf As New PdfDocument()

            ' Load an existing PDF document
            pdf.LoadFromFile("..\..\..\..\..\..\Data\FreeTextAnnotation.pdf")

            ' Get the first page of the PDF
            Dim firstPage As PdfPageBase = pdf.Pages(0)

            ' Create a new PdfDocument object to store the copied annotations
            Dim newPdf As New PdfDocument()

            ' Iterate through each annotation in the first page
            For Each annotation As PdfAnnotation In firstPage.Annotations.List

                ' Check if the annotation is a free text annotation
                If TypeOf annotation Is PdfFreeTextAnnotationWidget Then

                    ' Convert the annotation to a PdfFreeTextAnnotationWidget
                    Dim textAnnotation As PdfFreeTextAnnotationWidget = TryCast(annotation, PdfFreeTextAnnotationWidget)

                    ' Extract the bounds and text of the annotation
                    Dim rect = textAnnotation.Bounds
                    Dim text = textAnnotation.Text

                    ' Create a new page in the new PDF document with the same size as the original page
                    Dim newPage As PdfPageBase = newPdf.Pages.Add(firstPage.Size)

                    ' Create a new free text annotation in the new page
                    Dim newAnnotation As New PdfFreeTextAnnotation(rect)
                    newAnnotation.Text = text
                    newAnnotation.CalloutLines = textAnnotation.CalloutLines
                    newAnnotation.LineEndingStyle = textAnnotation.LineEndingStyle
                    newAnnotation.AnnotationIntent = PdfAnnotationIntent.FreeTextCallout
                    newAnnotation.RectangleDifferences = textAnnotation.RectangularDifferenceArray
                    newAnnotation.Color = textAnnotation.Color

                    ' Add the new annotation to the new page
                    newPage.Annotations.Add(newAnnotation)
                End If
            Next annotation

            ' Specify the output file path for the modified PDF
            Dim result As String = "CopyTextAnnotationProperties.pdf"

            ' Save the modified PDF document to the output file
            newPdf.SaveToFile(result)

            ' Close the PDF documents
            newPdf.Close()
            pdf.Close()

            ' Launch the file
            DocumentViewer(result)
        End Sub

        Private Sub DocumentViewer(ByVal filename As String)
            Try
                Process.Start(filename)
            Catch
            End Try
        End Sub
    End Class
End Namespace
